src/components/atoms/Badge/Badge.tsx

```diff
@@ -13,23 +15,39 @@ interface IBadgeProps extends Omit<HTMLAttributes<HTMLSpanElement>, 'color'> {
   styling?: TStyling;
 }
 
-const backgroundColors: IBgColors = {
-  confirmed: colors.successLight,
-  default: colors.greyLight,
-  invalid: colors.alertLight,
-  waiting: colors.warningLight,
+const colorMap: TColorMap = {
+  confirmed: {
+    bg: colors.successLight,
+    text: colors.success,
+  },
+  default: {
+    bg: colors.greyLight,
+    text: colors.greyDarkest,
+  },
+  invalid: {
+    bg: colors.alertLight,
+    text: colors.alertDark,
+  },
+  waiting: {
+    bg: colors.warningLight,
+    text: colors.warningDark,
+  },
 };
 
 const StyledBadge = styled(Text).attrs({
   size: 'small',
   forwardedAs: 'span',
 })<IBadgeProps>`
-  color: ${colors.greyDarkest};
-  background-color: ${({ styling = 'default' }) => styling && backgroundColors[styling]};
+  ${({ styling = 'default' }) => css`
+    background-color: ${colorMap[styling].bg};
+    color: ${colorMap[styling].text};
+  `};
+
   display: inline-block;
-  padding: 4px 10px;
+  padding: 3px 10px;
   white-space: nowrap;
-  border-radius: 4px;
+  border-radius: 12px;
+  font-weight: ${typography.weights.semiBold};
```

Comment:
do we need to add font-family?

Reply:
I think no as the `font-family` is declared globally in `body {}`